### DISCUSSION TITLE
Documentation on saving json

### DISCUSSION THREAD

**Discussion by bard-hobson** [react-page/react-page#976] [Q&A] [answered]
Love the idea of the editor, but I do not see any documentation on actually saving the json and then rendering it later as its own. The read-only button works but what about if I just want to rerender it on another page, ie as its own blog post?

**Comment by macrozone** [comment, marked as answer]
hi @bard-hobson 

there is an onChange callback which is called with the current value. You can save that to any endpoint.

Typically you will have something like this: https://react-page.github.io/docs/#/quick-start?id=simple-example

you could e.g. have a save button:

```

const YourComponent = () => {
 const [value, setValue] = useState<Value>(null);

 const saveValue = () => sendToDb(value) // this could be a graphql mutation, a rest call using fetch, save to firebase or whatever you need

  return ( 
   <div>
      <Editor cellPlugins={cellPlugins} value={value} onChange={setValue} />
      <Button onClick={saveValue}>Save</Button>
  </div>
  );
}

```


Basically, `<Editor />` is like a form-field. it has some value and can update a value.

If you want to just display the content, you use the editor in readOnly mode:

```
const value = // loaded through some api, e.g. graphql-query, fetch, etc.

return <Editor cellPlugins={cellPlugins} value={value}  />


```

**Reply by bard-hobson** [reply]
@macrozone 

Thanks for the quick reply.

Your answer is what I assumed, but did not work (probably I am missing something). I used the example json as a hard coded value to do a simple test. 

```
const value = '{"id":"obknih","version":1,"rows":[{"id":"b27eia","cells":[{"id":"e9htzt","size":12,"plugin":{"id":"ory/editor/core/content/slate","version":1},"dataI18n":{"default":{"slate":[{"type":"HEADINGS/HEADING-TWO","children":[{"text":"This is a heading"}]},{"type":"PARAGRAPH/PARAGRAPH","children":[{"text":"This is some paragraph text"}]}]}},"rows":[],"inline":null}]}]}';

 <Editor cellPlugins={cellPlugins} value={value} readOnly />
```

**Reply by bard-hobson** [reply]
I found my issue, put the json in single quotes which made the value a string instead of the json. Need to walk away for awhile lol.

**Reply by macrozone** [reply]
i guess typescript would warn you ;-) highly recommend
